flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalSortLimit.scala

```diff
@@ -113,35 +104,19 @@ class BatchExecSortLimit(
 
   //~ ExecNode methods -----------------------------------------------------------
 
-  override def getInputEdges: util.List[ExecEdge] = List(
-    ExecEdge.builder()
-      .damBehavior(ExecEdge.DamBehavior.END_INPUT)
-      .build())
-
-  override protected def translateToPlanInternal(
-      planner: BatchPlanner): Transformation[RowData] = {
-    if (limitEnd == Long.MaxValue) {
-      throw new TableException("Not support limitEnd is max value now!")
-    }
-
-    val input = getInputNodes.get(0).translateToPlan(planner)
-        .asInstanceOf[Transformation[RowData]]
-    val inputType = input.getOutputType.asInstanceOf[InternalTypeInfo[RowData]]
-    val types = inputType.toRowFieldTypes
-
-    // generate comparator
-    val genComparator = ComparatorCodeGenerator.gen(
-      planner.getTableConfig, "SortLimitComparator", keys, keys.map(types(_)), orders, nullsIsLast)
-
-    // TODO If input is ordered, there is no need to use the heap.
-    val operator = new SortLimitOperator(isGlobal, limitStart, limitEnd, genComparator)
-
-    ExecNodeUtil.createOneInputTransformation(
-      input,
-      getRelDetailedDescription,
-      SimpleOperatorFactory.of(operator),
-      inputType,
-      input.getParallelism,
-      0)
+  override def translateToExecNode(): ExecNode[_] = {
+    new BatchExecSortLimit(
```

Comment:
move `private val (keys, orders, nullsIsLast) = SortUtil.getKeysAndOrders(
    sortCollation.getFieldCollations)` into this method